src/components/sprint-view-table/tasks-table.tsx

```diff
@@ -0,0 +1,160 @@
+import { Box, Card, CircularProgress, IconButton, Typography } from "@mui/material";
+import { Projects, Tasks, TimeEntries } from "../../generated/homeLambdasClient";
+import { DataGrid } from "@mui/x-data-grid";
+import { useEffect, useState } from "react";
+import { useLambdasApi } from "../../hooks/use-api";
+import KeyboardArrowUpIcon from "@mui/icons-material/KeyboardArrowUp";
+import KeyboardArrowDownIcon from "@mui/icons-material/KeyboardArrowDown";
+import strings from "../../localization/strings";
+import sprintViewTasksColumns from "./sprint-tasks-columns";
+import { errorAtom } from "../../atoms/error";
+import { useSetAtom } from "jotai";
+
+/**
+ * Component properties
+ */
+interface Props {
+  project : Projects,
+  loggedInpersonId?: number,
+  filter?: string
+}
+
+/**
+ * Task component
+ * 
+ * @param props component properties
+ */
+const TaskTable = ({project, loggedInpersonId, filter}: Props) => {
+  const { tasksApi, timeEntriesApi } = useLambdasApi();
+  const [tasks, setTasks] = useState<Tasks[]>([]);
+  const [timeEntries, setTimeEntries] = useState<number[]>([]);
+  const [open, setOpen] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const columns = sprintViewTasksColumns({tasks, timeEntries});
+  const setError = useSetAtom(errorAtom);
+  const [reload, setReload] = useState(false);
+
+  /**
+   * Gather tasks and time entries when project is available and update changes appeared
+   */
+  useEffect(() => {
+    if (project && open) {
+      getTasksAndTimeEntries();
+    }
+  }, [project, open, filter, reload]);
+
+  /**
+   * Handle loggenInPersonId change
+   */
+  useEffect(()=>{
+    setReload(true);
+  },[loggedInpersonId]);
+  
+  /**
+   * Get tasks and total time entries  
+   */
+  const getTasksAndTimeEntries = async () => {
+    setLoading(true);
+    if (reload || !tasks.length || !timeEntries.length){
+      try {
+        let fetchedTasks = await tasksApi.listProjectTasks({projectId: project.id});
```

Comment:
Typically rather than have fetchedTasks as a let and then update it after the filter, fetchedTasks would be a const and then when we filter we create another const e.g. loggedInPersonsTasks (might be an appropriate name in this case)